runtime/src/bank/address_lookup_table.rs

```diff
@@ -51,22 +51,20 @@ impl Bank {
             .map_err(|_| AddressLoaderError::SlotHashesSysvarNotFound)?;
 
         let mut deactivation_slot = u64::MAX;
-        let loaded_addresses = address_table_lookups
-            .map(|address_table_lookup| {
+        let mut loaded_addresses = LoadedAddresses::default();
```

Comment:
We could, with some work in higher-level functions, pre-allocate the space for this.